storage/src/checker/mod.rs

```diff
@@ -261,6 +259,16 @@ impl<S: WritableStorage> NodeStore<Committed, S> {
         let (area_index, area_size) = self.area_index_and_size(subtrie_root_address)?;
         let (node, node_bytes) = self.read_node_with_num_bytes_from_disk(subtrie_root_address)?;
 
+        // check if the node fits in the area
+        if node_bytes > area_size {
```

Comment:
This feels like an off by one error but might be okay. Is it possible to remove the impossible below and just figure out what the minimum AreaIndex is first?

Reply:
This is not off by one but a pretty significant difference - see #1184